### PR TITLE
Removed recommonmark from docs build

### PR DESCRIPTION
**Pull request by eemperor**
Removed recommonmark from the docs build and updated associated markdown files to directly use the existing restructuredtext format.

### CODE REVIEW

**Comment by KevinPlus3** [comment]
Just a minor fix needed due to flake failure.  Do you have a snapshot of the before and after of the documentation?  Was wondering if they look the same or if they look different when using m2r as opposed to recommonmark.

**Comment by eemperor** [comment]
Attached are couple comparisons between my local files and what's on Readthedocs currently.  I couldn't tell a difference.
screen shot 2017-12-11 at 12 30 29 pm

screen shot 2017-12-11 at 12 32 32 pm

**Comment by lorengordon** [comment]
@eemperor This looks good, the docs are generating properly. One more thing to look into would be to switch as much of the rst syntax over to md syntax as we can.

Things like:

```
:doc:`configuration file <configuration>`
```

Can become: 

```
[configuration file](configuration.html)
```

And, also, the two `.rst` files can now be changed to `.md` files.

File: docs/faq.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ to looking for pypi packages signed with MD5 signatures. If you've enabled FIPS
 (or are using a build that has FIPS pre-enabled), MD5 is disabled in the kernel
 (due to being a weak hashing-method). You can either disable FIPS (not
 recommended) or explicitly force `pip` to use a different signature-index. The
-latter is detailed in the Linux section of the :doc:`usage <usage>` document.
+latter is detailed in the Linux section of the [usage] (usage.html) document.
```
Comment (lorengordon) [inline docs/faq.md]: Oops, got a space in there between the bracket and paren: `] (`

Open a new pr to fix?

File: docs/faq.md [hunk 2]
```diff
@@ -52,7 +52,7 @@ label may not be related to the error you are encountering.
 
 ## Does watchmaker support Enterprise Linux 7?
 
-Watchmaker is supported on RedHat 7 and CentOS 7. See the :doc:`index <index>`
+Watchmaker is supported on RedHat 7 and CentOS 7. See the [index] (index.html)
```
Comment (lorengordon) [inline docs/faq.md]: space is bad here too

**Pull request merged**